### ISSUE TITLE
zero result when use tensorrt build t5encoder

### ISSUE THREAD

**Issue by zhangvia** [NVIDIA/TensorRT#4485]
## Description

i 'm using tensorrt 10.9.0.34 to build t5encoder on rtx4090 with float16 precision
but it seems the result of .plan was zero tensor not nan not inf but zero. but the onnx model result was normal

## Environment

**TensorRT Version**: 10.9.0.34

**NVIDIA GPU**: rtx 4090

**NVIDIA Driver Version**: 535.161.07

**CUDA Version**: 12.4

**CUDNN Version**: 9.2.0.82

Operating System:

Python Version (if applicable): 3.11.11

PyTorch Version (if applicable):2.6.0

## Relevant Files

i 'm using flux fill text_encoder_2 weights to convert. i'm uploading the onnx weight to google drive [t5 encoder](https://drive.google.com/drive/folders/1Fug4OufXY2gMjn5F8mLd7-HKXkKg2Ofw?usp=sharing)

**Model link**:

## Steps To Reproduce
the script that i used to convert model to onnx is below
```
root_path = "huggingface/black-forest-labs/FLUX.1-Fill-dev/"
def export_onnx(model,onnx_path,input_dict,dynamic_axes,output_names):
    model.eval()
    if not os.path.exists(os.path.dirname(onnx_path)):
        os.makedirs(os.path.dirname(onnx_path))
    torch.onnx.export(
        model,
        tuple(input_dict.values()),
        onnx_path,
        input_names=list(input_dict.keys()),
        output_names=output_names,
        dynamic_axes=dynamic_axes,
        opset_version=17,
        do_constant_folding=True,
    )

def export_t5encoder():
    prompt = "Draw a dog"
    tokenizer = T5TokenizerFast.from_pretrained(os.path.join(root_path, "tokenizer_2"))
    text_encoder = T5EncoderModel.from_pretrained(os.path.join(root_path, "text_encoder_2"), torch_dtype=torch.float16)
    text_input = tokenizer(
        prompt,
        padding="max_length",
        max_length=tokenizer.model_max_length,
        truncation=True,
        return_tensors="pt",
    )
    print(text_input.input_ids.shape)
    original_forward = text_encoder.forward
    def custom_forward(self, input_ids, output_hidden_states, return_dict=False):
        outputs = original_forward(input_ids=input_ids, output_hidden_states=output_hidden_states, return_dict=return_dict)
        last_hidden_state = outputs[0]
        return last_hidden_state
    
    text_encoder.forward = types.MethodType(custom_forward, text_encoder)

    input_dict = {"input_ids": text_input.input_ids,"output_hidden_states":False}
    dynamic_axes = {"input_ids": {0: "batch", 1: "sequence_len"}}
    output_names = ["last_hidden_state"]
    onnx_path = f"{onnx_export_path}t5encoder_fp16/model_16.onnx"
    export_onnx(text_encoder, onnx_path, input_dict, dynamic_axes, output_names)
```
```
#!/bin/bash

onnx=model_16.onnx  
min_b=1
max_b=20  
seq_len=512

while (( min_b < max_b )); do
    mid_b=$(( (min_b + max_b + 1) / 2 ))

    echo "⏳ Testing batch=${mid_b} ..."
    trtexec \
        --onnx=${onnx} \
        --minShapes=input_ids:1x${seq_len} \
        --optShapes=input_ids:${mid_b}x${seq_len} \
        --maxShapes=input_ids:${mid_b}x${seq_len} \
        --memPoolSize=workspace:24576    \
        --fp16 \
        --device=1 \
        --saveEngine=model${mid_b}.plan \
        --verbose \
        &> log_bs${mid_b}.txt

    if [ $? -eq 0 ]; then
        echo "✅ Success at batch=${mid_b}"
        min_b=${mid_b}
    else
        echo "❌ OOM or build fail at batch=${mid_b}"
        max_b=$((mid_b - 1))
    fi
done

echo "🚀 Max buildable batch size: ${min_b}"
```
the onnx model can run without any errors:

![Image](https://github.com/user-attachments/assets/3edd1452-aad0-43da-b5c6-5b824dee9f23)
 
as you can see, the result of .plan was completely  a zero tensor

**Comment by LeoZDong** [comment]
I was able to reproduce this. The issue is that the `--fp16` flag is not meant to be used for running an FP16 model. Instead, the flag is for enabling FP16 kernels while running the model in **weakly typed mode**. Weakly typed mode + `--fp16` flag means that TensorRT will not always respect the precision of the original model, but will run any layer in FP16 if it is timed to be faster, _regardless of accuracy_.

Modern neural networks are sensitive to extreme intermediate values that might fall outside the FP16 range, so using weakly typed mode and allowing the entire network to run in FP16 will likely give you accuracy errors. In fact, we are deprecating weak typing in TensorRT 10.12. You can read more about strong typing vs. weak typing [here](https://docs.nvidia.com/deeplearning/tensorrt/latest/architecture/capabilities.html#strong-typing-vs-weak-typing).

In your case, I've inspected your model and see that it has already been _selectively_ converted into FP16 (many, but not all, layers are in FP16 already):
<img src="https://github.com/user-attachments/assets/41b59515-d52e-49e1-9aa0-bbeb2fc3e844" width="70%">

This means that in order to get the FP16 performance, you can simply run the model in strongly-typed mode (i.e. `--stronglyTyped` in `trtexec` or `NetworkFromOnnxPath(onnx_path, strongly_typed=True)` in `Polygraphy`). This gives the same behavior as simply running the model through ONNX Runtime in your repro.

I've tried comparing the running the model using strongly-typed TensorRT and ONNX Runtime, and see that the mean / median absolute difference is within `1e-3` with occasional outliers (i.e. max absolute difference), which is within the expected range for FP16. I've attached the script [here](https://gist.github.com/LeoZDong/55f516d871b062a56819306274f2eebc) for your reference.

Closing the issue for now. Feel free to followup or open a new issue if you have additional questions.
